File: java/src/main/java/com/molizhishu/demo/client/MolizhishuClient.java
package com.molizhishu.demo.client;

import com.molizhishu.demo.config.MolizhishuProperties;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.http.HttpHeaders;
import org.springframework.http.HttpMethod;
import org.springframework.http.MediaType;
import org.springframework.stereotype.Component;
import org.springframework.web.client.RestClient;

import java.time.Duration;
import java.time.Instant;
import java.util.HashMap;
import java.util.Map;
import java.util.Objects;

@Component
public class MolizhishuClient {

	private final MolizhishuProperties properties;

	private final RestClient restClient;

	private static final Logger log = LoggerFactory.getLogger(MolizhishuClient.class);

	public MolizhishuClient(MolizhishuProperties properties, RestClient.Builder builder) {
		this.properties = properties;
		this.restClient = builder.build();
	}

	public Map<String, Object> submitTask(Map<String, Object> payload) {
		return request(HttpMethod.POST, properties.baseUrl() + "/task/batch/shared", payload, "local-api:submit-task");
	}

	public Map<String, Object> getTaskStatus(String taskId, String source) {
		return request(HttpMethod.GET, properties.baseUrl() + "/task/status/" + taskId, null, source);
	}

	public Map<String, Object> getTaskResult(String taskId, String source) {
		return request(HttpMethod.GET, properties.baseUrl() + "/task/result/" + taskId, null, source);
	}

	public Object stopTask(String taskId) {
		return request(HttpMethod.PUT, properties.baseUrl() + "/task/" + taskId + "/stop", null, "local-api:stop-task").get("value");
	}

	public Object getCallbackUrl() {
		return request(HttpMethod.GET, properties.baseUrl() + "/task/callback-url", null, "local-api:callback-url:get").get("value");
	}

	public Object updateCallbackUrl(String callbackUrl) {
		Map<String, Object> payload = new HashMap<>();
		payload.put("callbackUrl", callbackUrl);
		return request(HttpMethod.PUT, properties.baseUrl() + "/task/callback-url", payload, "local-api:callback-url:update").get("value");
	}

	public Object getCities() {
		return request(HttpMethod.GET, properties.cityUrl(), null, "local-api:cities").get("value");
	}

	@SuppressWarnings("unchecked")
	private Map<String, Object> request(HttpMethod method, String url, Object payload, String source) {
		if (properties.token() == null || properties.token().isBlank()) {
			throw new MolizhishuApiException("MOLIZHISHU_TOKEN 未配置", null, 0);
		}

		Instant started = Instant.now();

		Map<String, Object> envelope = restClient.method(method)
				.uri(url)
				.header(HttpHeaders.AUTHORIZATION, "Bearer " + properties.token())
				.accept(MediaType.APPLICATION_JSON)
				.contentType(MediaType.APPLICATION_JSON)
				.body(payload == null ? "" : payload)
				.retrieve()
				.body(Map.class);

		boolean success = Boolean.TRUE.equals(Objects.requireNonNull(envelope).get("success"));
		Integer code    = envelope.get("code") instanceof Number number ? number.intValue() : null;
		String  message = envelope.get("message") == null ? "" : envelope.get("message").toString();

		log.info("[molizhishu] source={} method={} url={} success={} code={} message=\"{}\" duration={}ms",
				source, method.name(), url, success, code, message, Duration.between(started, Instant.now()).toMillis()
		);

		if (!success) {
			throw new MolizhishuApiException(message.isBlank() ? "模力指数业务处理失败" : message, code, 502);
		}

		Object data = envelope.get("data");

		if (data instanceof Map<?, ?> map) {
			return (Map<String, Object>) map;
		}

		return Map.of("value", data);
	}

}

File: dotnet/Client/MolizhishuClient.cs
using System.Text;
using System.Text.Json.Nodes;
using Molizhishu.ApiPub.Config;
using Molizhishu.ApiPub.Support;

namespace Molizhishu.ApiPub.Client;

/// <summary>
/// Typed HTTP client for Molizhishu monitor APIs.
/// </summary>
public sealed class MolizhishuClient
{
    private readonly HttpClient _http;
    private readonly AppSettings _settings;
    private readonly ILogger<MolizhishuClient> _logger;

    public MolizhishuClient(HttpClient http, AppSettings settings, ILogger<MolizhishuClient> logger)
    {
        _http = http;
        _settings = settings;
        _logger = logger;
        _http.Timeout = TimeSpan.FromSeconds(settings.TimeoutSeconds);
    }

    public Task<JsonObject> SubmitTaskAsync(JsonObject payload) => RequestAsync(HttpMethod.Post, $"{_settings.BaseUrl}/task/batch/shared", payload, "local-api:submit-task");
    public Task<JsonObject> GetTaskStatusAsync(string taskId, string source) => RequestAsync(HttpMethod.Get, $"{_settings.BaseUrl}/task/status/{Uri.EscapeDataString(taskId)}", null, source);
    public Task<JsonObject> GetTaskResultAsync(string taskId, string source) => RequestAsync(HttpMethod.Get, $"{_settings.BaseUrl}/task/result/{Uri.EscapeDataString(taskId)}", null, source);
    public async Task<JsonNode?> StopTaskAsync(string taskId) => (await RequestAsync(HttpMethod.Put, $"{_settings.BaseUrl}/task/{Uri.EscapeDataString(taskId)}/stop", null, "local-api:stop-task"))["value"];
    public async Task<JsonNode?> GetCallbackUrlAsync() => (await RequestAsync(HttpMethod.Get, $"{_settings.BaseUrl}/task/callback-url", null, "local-api:callback-url:get"))["value"];
    public async Task<JsonNode?> UpdateCallbackUrlAsync(string? callbackUrl) => (await RequestAsync(HttpMethod.Put, $"{_settings.BaseUrl}/task/callback-url", new JsonObject { ["callbackUrl"] = callbackUrl }, "local-api:callback-url:update"))["value"];
    public async Task<JsonNode?> GetCitiesAsync() => (await RequestAsync(HttpMethod.Get, _settings.CityUrl, null, "local-api:cities"))["value"];

    private async Task<JsonObject> RequestAsync(HttpMethod method, string url, JsonObject? payload, string source)
    {
        if (string.IsNullOrWhiteSpace(_settings.Token)) throw new ApiException("MOLIZHISHU_TOKEN 未配置", null, 502);

        using var request = new HttpRequestMessage(method, url);
        request.Headers.Authorization = new System.Net.Http.Headers.AuthenticationHeaderValue("Bearer", _settings.Token);
        request.Headers.Accept.ParseAdd("application/json");
        if (payload is not null) request.Content = new StringContent(payload.ToJsonString(), Encoding.UTF8, "application/json");

        var started = DateTimeOffset.UtcNow;
        using var response = await _http.SendAsync(request);
        var body = await response.Content.ReadAsStringAsync();
        var envelope = JsonNode.Parse(body)?.AsObject() ?? throw new ApiException("模力指数接口返回非 JSON", null, (int)response.StatusCode);
        var success = envelope.BoolValue("success");
        var code = envelope.IntValue("code");
        var message = envelope.StringValue("message");

        _logger.LogInformation("[molizhishu] source={Source} method={Method} url={Url} http_status={Status} success={Success} code={Code} message=\"{Message}\" duration={Duration}ms",
            source, method.Method, url, (int)response.StatusCode, success, code, message, (DateTimeOffset.UtcNow - started).TotalMilliseconds);

        if (!response.IsSuccessStatusCode) throw new ApiException($"模力指数 HTTP 异常：{(int)response.StatusCode}", code, (int)response.StatusCode);
        if (!success) throw new ApiException(string.IsNullOrWhiteSpace(message) ? "模力指数业务处理失败" : message, code, 502);

        var data = envelope["data"];
        return data is JsonObject obj ? obj.DeepClone().AsObject() : new JsonObject { ["value"] = data?.DeepClone() };
    }
}
